routing/matcher.go

```diff
@@ -235,8 +247,14 @@ func newMatcher(rs []*Route, o MatchingOptions) (*matcher, []*definitionError) {
 }
 
 // matches a path in the path trie structure.
-func matchPathTree(tree *pathmux.Tree, path string) (leafMatchers, map[string]string) {
-	v, params := tree.Lookup(path)
+func matchPathTree(tree *pathmux.Tree, path string, lrm *leafRequestMatcher) (leafMatchers, map[string]string) {
+	var matcher pathmux.Matcher = nil
```

Comment:
this strange thing needed, because in Tree somehow m != nil check true for nil. Perhaps bug in Go?